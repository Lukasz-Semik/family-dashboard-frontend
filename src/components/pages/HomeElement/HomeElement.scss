.title-wrapper {
  margin: 0.3rem 0 2rem 1rem;
}

.links-wrapper {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 60rem;
  max-width: 92%;
  margin: 0 auto 1rem;
  transform: scaleX(1);
}

.toggle-view-leave-active,
.toggle-view-enter-active {
  transition: opacity 0.2s ease-in-out, transform 0.2s ease-in-out;
}

.toggle-view-leave-to,
.toggle-view-enter {
  opacity: 0;
  transform: scaleX(0);
}
