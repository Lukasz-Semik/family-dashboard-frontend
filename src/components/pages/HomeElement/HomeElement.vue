<template>
  <div>
    <div :class="[$style['title-wrapper']]">
      <TitleElement translation-path="general.appTitle"/>
    </div>

    <div :class="[$style['links-wrapper']]">
      <Link :to="signInRoute" translation-path="general.signIn"/>
      <Link :to="signUpRoute" translation-path="general.signUp"/>
    </div>

    <Transition
      :leave-active-class="$style['toggle-view-leave-active']"
      :leave-class="$style['toggle-view-leave']"
      :enter-active-class="$style['toggle-view-enter-active']"
      :enter-class="$style['toggle-view-enter']"
      mode="out-in"
    >
      <RouterView/>
    </Transition>
  </div>
</template>

<script>
import TitleElement from '@/components/atoms/TitleElement/TitleElement.vue';
import Link from '@/components/atoms/Link/Link.vue';
import { SIGN_IN_ROUTE, SIGN_UP_ROUTE } from '@/constants/routesNames';

export default {
  components: {
    TitleElement,
    Link,
  },
  data() {
    return {
      signInRoute: { name: SIGN_IN_ROUTE },
      signUpRoute: { name: SIGN_UP_ROUTE },
    };
  },
};
</script>

<style lang="scss" module src="./HomeElement.scss" />
