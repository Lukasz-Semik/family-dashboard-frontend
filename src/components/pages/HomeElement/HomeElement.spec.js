import { shallowMount } from '@vue/test-utils';
import Vue from 'vue';
import VueRouter from 'vue-router';

import HomeElement from './HomeElement.vue';

Vue.use(VueRouter);

describe('<HomeElement/>', () => {
  it('should render properly', () => {
    const wrapper = shallowMount(HomeElement);

    expect(wrapper).toMatchSnapshot();
  });
});
