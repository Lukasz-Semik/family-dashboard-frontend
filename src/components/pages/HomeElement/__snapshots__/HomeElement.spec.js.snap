// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeElement/> should render properly 1`] = `
<div>
  <toastelement-stub translated-text="some fucking notification"></toastelement-stub>
  <div class="title-wrapper">
    <titleelement-stub tag="h1" translatedtext="" translationpath="general.appTitle"></titleelement-stub>
  </div>
  <div class="links-wrapper">
    <link-stub to="[object Object]" translatedtext="" translationpath="general.signIn"></link-stub>
    <link-stub to="[object Object]" translatedtext="" translationpath="general.signUp"></link-stub>
  </div>
  <routerview-stub name="default"></routerview-stub>
</div>
`;
